decentraland/redux/reducers.js

```diff
@@ -5,43 +5,31 @@ import {
   SET_LAND_FOR_SALE_LIST,
   REMOVE_LAND_FOR_SALE,
   ADD_LAND_FOR_SALE_TO_LIST,
-  SET_SETUP_IN_PROGRESS,
-  SET_ACCOUNT_FUNDED_WITH_ETHERS,
-  SET_ACCOUNT_FUNDED_WITH_MANA,
-  SET_ACCOUNT_APPROVED_MARKETPLACE,
+  SET_ACCOUNT_CREATION_STATUS,
   SET_LOADING_ASSETS_FOR_SALE_IN_PROGRESS,
   ADD_TO_MY_ASSETS_LIST,
 } from "./actions";
 
+import AccountCreationStatus from "@constants/AccountCreationStatus";
+const { NOT_STARTED } = AccountCreationStatus;
+
 function accountInfo(
   state = {
     account: null,
     setupInProgress: false,
     fundedWithEthers: false,
     fundedWithMana: false,
     marketplaceApproved: false,
+    creationStatus: NOT_STARTED,
   },
   action
 ) {
-  const {
-    type,
-    account,
-    inProgress: setupInProgress,
-    fundedWithEthers,
-    fundedWithMana,
-    approvedMarketplace,
-  } = action;
+  const { type, account, creationStatus } = action;
   switch (type) {
     case SET_ACCOUNT:
       return { ...state, account };
-    case SET_SETUP_IN_PROGRESS:
-      return { ...state, setupInProgress };
-    case SET_ACCOUNT_FUNDED_WITH_ETHERS:
-      return { ...state, fundedWithEthers };
-    case SET_ACCOUNT_FUNDED_WITH_MANA:
-      return { ...state, fundedWithMana };
-    case SET_ACCOUNT_APPROVED_MARKETPLACE:
-      return { ...state, approvedMarketplace };
+    case SET_ACCOUNT_CREATION_STATUS:
+      return { ...state, creationStatus };
```

Comment:
Would this be where we would want to make a future PR enforcing rules about allowed state transitions?

Reply:
@marcelomorgado Curious for your thoughts on this when you get a chance. Thanks!

Reply:
Hey @pcowgill, it isn't an easy question, I've read a bit about it (where put business logic code on react apps). 
Using `redux` we have some options for it: `actions`, `reducers`, `thunk`/`saga`, our own `middlewares` or a specific lib, each one with pros/cons. The app scale planning seems the major concern on how the solution to use. For our current needs I think that we can use `actions` or `reducers` to that and move to `thunk`/`saga` if/when will add it to the project.

I haven't found articles about business logic using `React Context` so I need to look deep into this to have an answer.

Refs:
https://redux.js.org/faq/code-structure#how-should-i-split-my-logic-between-reducers-and-action-creators-where-should-my-business-logic-go
https://medium.com/@jeffbski/where-do-i-put-my-business-logic-in-a-react-redux-application-9253ef91ce1

Reply:
Thanks for looking into this! That all makes sense.

I think even if we used `thunk` or `saga` it could still make sense to enforce state transitions based on the returned data outside the scope of the async dispatch that the `thunk` or `saga` would take over.

Your suggestion of how to handle it for now sounds good. In particular I think it makes sense in the reducer based on this wording `"Reducers specify how the application's state changes in response to actions sent to the store"` ([Source](https://redux.js.org/basics/reducers))

Yeah, we can think about how `React Context` fits in maybe for `0.2.0`

Custom middleware or an open-source middleware lib might make sense if we ever want to start enforcing the same transition rules on multiple reducers

Reply:
So long story short, I think my personal answer to my original question

> Would this be where we would want to make a future PR enforcing rules about allowed state transitions?

is yes. Does that work for you?

Reply:
Yes, I agree :+1:

Reply:
Okay, I'll resolve this and create an issue for this.

Reply:
https://github.com/tasitlabs/tasit/issues/239